xenforo-scraper.py

```diff
@@ -232,23 +240,31 @@ def scrapepage(url):
         if args.debug:
             print("Saving to:", fullpath)
 
+        import pdb
         if os.path.exists(fullpath):
-            print(f"\x1b[KProgress: {count}/{len(files)} - Skipping {truncated}", end="\r")
+            filename, extension = os.path.splitext(fullpath)
+            counter = 1
+            while os.path.exists(fullpath):
+                if os.path.getsize(fullpath) == filesize:
+                    print(f"\n[KProgress: {count}/{len(files)} - Skipping {truncated}", end="\r")
+                    break
+                fullpath = f"{filename} + ({counter}){extension}"
+                counter += 1
             continue
```

Comment:
This part will never make a difference as the continue under this will always skip the file. Also the fullpath is incorrect as it will literally add a + sign in the middle of the filename.

It would need to be something along the lines of:

```suggestion
            filename, extension = os.path.splitext(fullpath)
            counter = 1
            while os.path.exists(fullpath):
                if os.path.getsize(fullpath) == filesize:
                    print(f"\n[KProgress: {count}/{len(files)} - Skipping {truncated}", end="\r")
                    break
                fullpath = f"{filename}({counter}){extension}"
                counter += 1
            if os.path.exists(fullpath): continue
```

Reply:
245-253 lines make sure that different files with same names will be saved not overwritting.